### PR TITLE
Admin Generator (Future): Add ability to hide columns depending on the breakpoint

### PR DESCRIPTION
**Pull request by jamesricky**
Demonstrate this feature by replacing the "Title" and "Description" columns with a single "Overview" column on smaller screens. 

---

## PR Checklist

-   [x] Merge parent PR https://github.com/vivid-planet/comet/pull/2491
-   [x] Verify if the change requires a changeset. See [CONTRIBUTING.md](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md)
-   [x] Link to the respective task if one exists: SVK-238
-   [x] Provide screenshots/screencasts if the change contains visual changes

| Above MD breakpoint (desktop) | Below MD breakpoint (mobile) |
| --- | --- |
| DataGrid above MD breakpoint | DataGrid below MD breakpoint|

### CODE REVIEW

File: demo/admin/src/products/future/ProductsGrid.cometGen.ts [hunk 1]
```diff
@@ -9,10 +9,20 @@ export const ProductsGrid: GridConfig<GQLProduct> = {
...
-        { type: "text", name: "description", headerName: "Description" },
+        {
+            type: "combination",
+            name: "overview",
+            headerName: "Overview",
+            minWidth: 200,
+            maxWidth: 250,
+            primaryText: "title",
+            secondaryText: "description",
+            visible: (breakpoints) => breakpoints.down("md"),
```
Comment (johnnyomair) [inline demo/admin/src/products/future/ProductsGrid.cometGen.ts:20]: Why is this a function? For type safety?

Comment (jamesricky) [reply demo/admin/src/products/future/ProductsGrid.cometGen.ts:20]: For type safety and for easy access to the themes breakpoints. 
This also allows setting the breakpoint in the same way you'd do in other places with MUI, e.g., the handmade grid:

https://github.com/vivid-planet/comet/blob/02a4da594177936cc243daea72a6df8cfa77df48/demo/admin/src/products/ProductsGrid.tsx#L80

An alternative would be to set `visible` to an object like this:

```ts
visible: {
    direction: "down",
    breakpoint: "md",
},
```
```ts
visible: {
    direction: "up",
    breakpoint: "md",
},
```
```ts
visible: {
    direction: "between",
    start: "md",
    end: "lg",
},
```

Comment (johnnyomair) [reply demo/admin/src/products/future/ProductsGrid.cometGen.ts:20]: We could also achieve type safety with advanced types: https://www.typescriptlang.org/play/?#code/C4TwDgpgBAIglgJwgY2HA9gOygXigIgFcx8oAfAgE3QHdNSL8AjCYGiCegbgFgAoUJCgBpCCFxQA5PgC2lfJPJT8AGwDmCpZkIyWCXn36DoANTgBnOExXQ8AA2AALCDIgA6JkgCGAazDo4TGBzNwASAG94JFQMTABfAAoI0RA4gEo7A35kLHNgKC8ALigzS2tbKScXd08IX39A4I9Wdk4E1Q00yWzc-KZi0qsbCUkq1w9vPwCgkJY2Dkx25xUVdHwuoA

@nsams what do you think? Is a function in the config okay?

Comment (jamesricky) [reply demo/admin/src/products/future/ProductsGrid.cometGen.ts:20]: > maybe like you advanced type but a bit [shorter](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAIglgJwgY2HA9gOygXigIgFcx8oAfAgE3QHdNSL8AjCYGiCegbgFgAoUJCgBpCCFxQA5PgC2lfJPJT8AGwDmCpZkIyWCXn36DoANTgBnOExXQ8AAwAkAb3hJUGTAF8AFM9EhPAEo7A35kLHNgKABDAC4oM0trWykWNg5Mb1UNQMkwiKimeMSrGwlJNPZOLIALCBUVdHxcoA)?

I like this 👍🏼

Comment (johnnyomair) [reply demo/admin/src/products/future/ProductsGrid.cometGen.ts:20]: @jamesricky please change it to the proposed solution.

Comment (jamesricky) [reply demo/admin/src/products/future/ProductsGrid.cometGen.ts]: Done in 2904df20fb3b6d5bf541830ca18298d5c4c5f0f7

**Review by nsams — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**